### PR TITLE
Avoid duplicate class symbol extraction

### PR DESCRIPTION
**Pull request by jpbempel**
# What Does This Do
We need to track all classes loaded and extracted into ClassNameTrie that is space efficient prefix tree. That way we prevent any duplicates reaching SymDB backend

# Motivation
Same classes can be loaded in different classloader, but symbol-wise they are the same. In some situation (like Groovy script) we can a lot
 of class loaded for same symbols which pollute SymDB backend.

# Additional Notes

# Contributor Checklist

- [ ] Format the title [according the contribution guidelines](https://github.com/DataDog/dd-trace-java/blob/master/CONTRIBUTING.md#title-format)
- [ ] Assign the `type:` and (`comp:` or `inst:`) labels in addition to [any usefull labels](https://github.com/DataDog/dd-trace-java/blob/master/CONTRIBUTING.md#labels)
- [ ] Squash your commits prior merging or merge using GitHub's [Squash and merge](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/incorporating-changes-from-a-pull-request/about-pull-request-merges#squash-and-merge-your-commits)
- [ ] Don't use `close`, `fix` or any [linking keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) when referencing an issue.  
  Use `solves` instead, and assign the PR [milestone](https://github.com/DataDog/dd-trace-java/milestones) to the issue
- [ ] Update the [public documentation](https://docs.datadoghq.com/tracing/trace_collection/library_config/java/) in case of new configuration flag or behavior

Jira ticket: [DEBUG-3114]

[DEBUG-3114]: https://datadoghq.atlassian.net/browse/DEBUG-3114?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: dd-java-agent/agent-debugger/src/main/java/com/datadog/debugger/symbol/SymbolAggregator.java [hunk 1]
```diff
@@ -61,10 +62,13 @@ public void parseClass(String className, byte[] classfileBuffer, String jarName)
     if (className.endsWith(CLASS_SUFFIX)) {
       className = className.substring(0, className.length() - CLASS_SUFFIX.length());
     }
-    Set<String> localLoadedClasses = loadedClasses;
-    if (localLoadedClasses != null && !localLoadedClasses.add(className)) {
-      // class already loaded and symbol extracted
-      return;
```
Comment (ojung) [inline dd-java-agent/agent-debugger/src/main/java/com/datadog/debugger/symbol/SymbolAggregator.java:67]: I don't understand: Shouldn't we have filtered duplicates with this code already? 🤔

Comment (jpbempel) [reply dd-java-agent/agent-debugger/src/main/java/com/datadog/debugger/symbol/SymbolAggregator.java:67]: only at reload time when we are processing already loaded classes.
after that we dropped the Set
Now it's permanent

File: dd-java-agent/agent-debugger/src/main/java/com/datadog/debugger/symbol/SymbolAggregator.java [hunk 2]
```diff
@@ -110,14 +114,4 @@ private void addJarScope(Scope jarScope, boolean forceFlush) {
...
-
-  void loadedClassesProcessStarted() {
-    // to avoid duplicate symbol extraction we keep track of loaded classes
-    // during the loaded class extraction phase
-    loadedClasses = ConcurrentHashMap.newKeySet();
-  }
-
-  void loadedClassesProcessEnded() {
-    loadedClasses = null;
-  }
```
Comment (ojung) [inline dd-java-agent/agent-debugger/src/main/java/com/datadog/debugger/symbol/SymbolAggregator.java:122]: Aha! I see now :) Thanks!

**Review by ojung — APPROVED** [review]
Thanks!

**Pull request merged**